gateway/gateway.go

```diff
@@ -1253,7 +1256,7 @@ func (gateway *HandleT) collectMetrics(ctx context.Context) {
 			case <-ctx.Done():
 				return
 			case <-gateway.diagnosisTicker.C:
-				gateway.requestMetricLock.RLock()
+				gateway.requestMetricLock.Lock()
```

Comment:
we are updating the metric so a lock is required instead of a read lock

Reply:
might as well convert it to a `Mutex` then, `RLock()` doesn't seem to be used anywhere else.

Reply:
yes indeed